src/BlockTypes/LegacyTemplate.php

```diff
@@ -180,4 +190,47 @@ protected function render_archive_product() {
 		wp_reset_postdata();
 		return ob_get_clean();
 	}
+
+	/**
+	 * Get HTML markup with the right classes by attributes.
+	 * This function appends the classname at the first element that have the class attribute.
+	 * Based on the experience, all the wrapper elements have a class attribute.
+	 *
+	 * @param string $content Block content.
+	 * @param array  $block Parsed block data.
+	 * @return string Rendered block type output.
+	 */
+	public function add_alignment_class_to_wrapper( string $content, array $block ) {
+		if ( ( 'woocommerce/' . $this->block_name ) !== $block['blockName'] ) {
+			return $content;
+		}
+
+		$attributes            = (array) $block['attrs'];
+		$align_class_and_style = StyleAttributesUtils::get_align_class_and_style( $attributes );
+
+		if ( ! isset( $align_class_and_style['class'] ) ) {
+			return $content;
+		}
+
+		// Find the first tag.
+		$first_tag = '<[^<>]+>';
+		$matches   = array();
+		preg_match( $first_tag, $content, $matches );
+
+		// If there is a tag, but it doesn't have a class attribute, add the class attribute.
+		if ( isset( $matches[0] ) && strpos( $matches[0], 'class' ) === false ) {
```

Comment:
We should check for the exact class attribute here. Elements can have `class` as a part of attribute name or attribute value, like:  `data-class`, `data-type="class"`, `id="fist-class"`.
```suggestion
		if ( isset( $matches[0] ) && strpos( $matches[0], ' class=' ) === false ) {
```